rfd/0xxx-tls-routing-behind-lb.md

```diff
@@ -0,0 +1,286 @@
+  │                       │  │local client credentials │
+  │                       │◄─┘local KUBECONFIG         │
+  │                       │                            │
+  │ server localhost:8888 │                            │
+  │ local credentials     │                            │
+  ├──────────────────────►│  server kube-xxx.xxx.xxx   │
+  │                       │  "real" credentials        │
+  │                       ├───────────────────────────►│
+  │                       │                            │
+  │                       │◄───────────────────────────┤
+  │◄──────────────────────┤                            │
+  │                       │                            │
+  │ another request       │                            │
+  ├──────────────────────►│                            |
+  │                       ├───────────────────────────►│
+  ...                   
+```
+
+Once the local server receives a request and verifies the request, the local server will proxy the request using the TLS
+Routing dialer to connect the Teleport Proxy. The local proxy should be in charge of managing the Teleport cert used for
+routing Kubernetes requests.
+
+Note that `kube-teleport-proxy-alpn.my-teleport-cluster.com` is not required to be resolvable (by DNS) in this case, but
+it will be used internally for routing.
+
+### User Experience
+
+Once the connection upgrade support to all protocols are implemented, users can be recommended to upgrade their Teleport
+clusters to single port mode if the cluster is currently serving separate ports.
+
+All scenarios discussed in this section assumes Teleport sits behind a load balancer so connection upgrade for TLS
+Routing is required.
+
+#### 1 - Database Access UX
+
+Common flows already use `tsh` local proxy (since v8) so they will automatically perform the connection upgrade:
+
+- `tsh db connect` (single port mode)
+- `tsh proxy db`
+
+So no UX change to these.
+
+In single port mode, using native clients to connect to Teleport Proxy directly with `tsh db env/config` will NOT work
+through load balancers. Users can use one of the above methods instead.
+
+#### 2 - Application Access UX
+
+Application Access through Teleport Webapp is NOT affected by the issue of interest of this RFD.
+
+On the `tsh` side, TCP, AWS, Azure, and GCP apps always use a local proxy so they will automatically perform the
+connection upgrade when needed. No UX change to these.
+
+However, HTTP apps MUST use `tsh proxy app`.
+
+#### 3 - Kubernetes Access UX
+
+Using native Kubernetes clients after `tsh kube login` will NOT work as the native clients cannot upgrade the
+connection.
+
+`tsh kubectl` will automatically starts a local proxy and performs the connection upgrade when needed.
+
+A new `tsh proxy kube` command will be added to support other Kubernetes clients. The command will start a local proxy
+and provide a config for Kubernetes clients to use:
+
+```
+$ tsh proxy kube -p 8888
```

Comment:
if the `tsh proxy kube` will also generate the `KUBECONFIG` it should also support kube cluster name parameter to allow a user to select a kube cluster where the request should be routed.

Reply:
added a paragraph on the flags.